ncm-network/src/main/perl/network.pm

```diff
@@ -1,1183 +1,1728 @@
+                $self->debug(1, "Replaced alias $name with alias name $new_name for interface $ifname");
+                $iface->{aliases}->{$new_name} = $alias;
+                delete $iface->{aliases}->{$name};
             }
-        } else {
-            $set_hwaddr = $set_hwaddr_default;
         }
 
-        if ($set_hwaddr) {
-            if (exists($net{$iface}{'hwaddr'})) {
-                $text .= "HWADDR=".$net{$iface}{'hwaddr'}."\n";
-            } else {
-                # huh?
-                $self->error("set_hwaddr is true and no hwaddr defined for device $iface.",
-                             " Bug in component. Please contact the developers.");
+        # bonding/bridging options
+        foreach my $attr (qw(bonding_opts bridging_opts)) {
+            my $opts = $iface->{$attr};
+            if (defined($opts) && keys %$opts) {
+                $iface->{$attr} = [map {"$_=$opts->{$_}"} sort keys %$opts];
+                $self->debug(1, "Replaced $attr with ", join(' ', @{$iface->{$attr}}), " for interface $ifname");
             }
         }
 
-        # set the networktype
-        if ( $net{$iface}{'mtu'} ) {
-            $text .= "MTU=".$net{$iface}{'mtu'}."\n";
+        # add ethtool options preparsed. These will be set in ifcfg- config
+        # some are needed on boot (like autoneg/speed/duplex)
+        if (exists($iface->{ethtool})) {
+            $iface->{ethtool_opts} = $self->ethtool_options($iface->{ethtool});
+            $self->debug(1, "Added ethtool_opts with ", join(' ', @{$iface->{ethtool}}), " for interface $ifname");
         }
 
-        # set the bootprotocol
-        my $bootproto;
-        if ( $net{$iface}{'bootproto'} ) {
-            $bootproto=$net{$iface}{'bootproto'};
-        } else {
-            $bootproto="static";
-        }
-        $text .= "BOOTPROTO=".$bootproto."\n";
+        # Handle hardware address
+        my $nicname = $ifname;
 
-        if ($bootproto eq "static") {
-            # set ipaddr
-            if ($net{$iface}{'ip'}) {
-                $text .= "IPADDR=".$net{$iface}{'ip'}."\n";
-            } else {
-                $self->error("Using static bootproto and no ",
-                             "ipaddress configured for $iface");
-            }
-            # set netmask
-            if ($net{$iface}{'netmask'}) {
-                $text .= "NETMASK=".$net{$iface}{'netmask'}."\n";
+        # TODO: can we get rid of this?
+        if (! exists($nics->{nicname}) and $ifname =~ m/^eth(\d+)/) {
+            # Try CERN nic as list
+            if (exists($nics->{1})) {
+                $nicname = $1 ;
+                $self->verbose("No nic found for $ifname, using nic-as-list name $nicname");
+            };
+        };
+
+
+        # Each iface has set_hwaddr
+        # If set_hwaddr is true, hwaddr must be set
+        my $nic = $nics->{$nicname} || {};
+        my $mac = $nic->{hwaddr};
+        $iface->{set_hwaddr} = $set_hwaddr_default;
```

Comment:
Doesn't this override the profile regardless?
Commenting it out restores the `HWADDR` line on my test node.

Reply:
what is your default set to in the profile? (i need to know if this is a bug or some backwards incompatible change).

Reply:
The default is set to false, see below.

Reply:
nvm, this is a stupid bug